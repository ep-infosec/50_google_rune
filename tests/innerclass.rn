//  Copyright 2021 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

class A(self, <value>) {
  class B(self, <value>) {
    self.value = value

    func dump(self) {
      println self.value
    }
  }

  self.b = self.B(value)
}

a1 = A(8u32)
a1.b.dump()
a2 = A("test")
a2.b.dump()
b1 = a1.B(-1i8)
b1.dump()
b2 = A.B(123i8)
b1 = b2
b1.dump()
