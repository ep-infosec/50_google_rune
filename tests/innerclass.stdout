8
test
-1
123
